### PR TITLE
COOP+COEP: more flaky test measures

### PR DESCRIPTION
**Pull request by annevk**
The infrastructure at https://web-platform-tests.org/writing-tests/testharness.html#variants seemed heavier weight than this homegrown solution. Maybe I did something wrong?

Helps with #21107.

### CODE REVIEW

File: html/cross-origin-opener-policy/coep-navigate-popup.https.html [hunk 1]
```diff
@@ -43,8 +46,14 @@
     "coep": "",
     "opener": false
   }
-].forEach(variant => {
+].forEach((variant, i) => {
   ["same-origin", "same-site"].forEach(site => {
+    // Only run specified variants
+    if (!window.location.search.substr(1).split(",").includes(i.toString())) {
```
Comment (zcorpan) [inline html/cross-origin-opener-policy/coep-navigate-popup.https.html]: I think I prefer using `subset-test.js` and `if (!shouldRunSubtest(i)) { return; }`

Although this is less code, it's missing some aspects like ability to run all tests when not using a query string and the "-last" thing which helps somewhat mitigate the risk of forgetting to update `<meta>`s when adding a new test.

Comment (annevk) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: Mkay.

Comment (annevk) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: I don't understand why that utility suggests 1-based indexing or how that even works given the code.

Also, given that I'm only doing this for the inner forEach loop there's a 2x difference with what I want. Sigh.

Comment (zcorpan) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: The first test is 1 when using `subsetTest()`. Using `shouldRunSubTest()` manually without `subsetTest()` it's up to you which numbers you use.

I don't follow about the 2x difference. Can you clarify?

Comment (zcorpan) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: Oh, I see. Can't you move it to the outer loop?

Comment (annevk) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: If I don't care about generating variants this is fine, right?

Comment (annevk) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: Outer loop would have the same problem btw, you'd need a counter across loops.

Comment (annevk) [reply html/cross-origin-opener-policy/coep-navigate-popup.https.html]: Oh, but I could move the check, yes.

**Review by zcorpan — APPROVED** [review]

**Pull request merged**